From: Michael Holzheu <holzheu@linux.vnet.ibm.com>
Subject: [PATCH] [BZ 89928] kdump: Do not add standby memory for kdump
Patch-mainline: not yet
Git-commit: -
References: bnc#809678, LTC#89928

Description:  kdump: Do not add standby memory for kdump
Symptom:      For systems that have defined standby memory the kdump
              kernel consumes additional memory. This can lead to
              an out-of-memory kernel panic if not enough crashkernel
              memory has been defined.
Problem:      The kdump kernel allocates page structures for the standby
              memory. This consumes 64 byte per page of standby memory.
Solution:     Do not use standby memory for the kdump kernel.
Reproduction: 1. Setup LPAR or z/VM with standby memory:
                 LPAR: Modify activation profile
                 z/VM: #cp def store XXXG standby YYYG
              2. Start Linux production system
              3. Verify standby memory with "lsmem" -> offline memory
              4. Load kdump kernel:
                 service kdump start
              5. Force kdump:
                 $ echo c > /proc/sysrq-trigger
              6. Login to kdump system (if possible)
              7. For kdump kernels without the fix you will see a line
                 "System RAM" in /proc/iomem for the the standy-memory:
                 $ cat /proc/iomem
                 ...
                 80000000-ffffffff : System RAM

Signed-off-by: Michael Holzheu <holzheu@linux.vnet.ibm.com>
Acked-by: John Jolly <jjolly@suse.de>
---
 drivers/s390/char/sclp_cmd.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/s390/char/sclp_cmd.c
+++ b/drivers/s390/char/sclp_cmd.c
@@ -611,6 +611,8 @@ static int __init sclp_detect_standby_me
 	struct read_storage_sccb *sccb;
 	int i, id, assigned, rc;
 
+	if (OLDMEM_BASE) /* No standby memory in kdump mode */
+		return 0;
 	if (!early_read_info_sccb_valid)
 		return 0;
 	if ((sclp_facilities & 0xe00000000000ULL) != 0xe00000000000ULL)
